src/core/coap/coap_message.cpp

```diff
@@ -146,8 +146,17 @@ uint8_t Message::WriteExtendedOptionField(uint16_t aValue, uint8_t *&aBuffer)
     return rval;
 }
 
-Error Message::AppendOption(uint16_t aNumber, uint16_t aLength, const void *aValue)
+Error Message::AppendOptionHeader(uint16_t aNumber, uint16_t aLength)
 {
+    /*
+     * Appends a CoAP Option header field (Option Delta/Length) per RFC 7252.
+     *
+     * @retval kErrorNone           Successfully appended the bytes.
+     * @retval kErrorInvalidArgs    The option type is not equal or greater than the last option type.
+     * @retval kErrorNoBufs         Insufficient available buffers to grow the message.
+     *
+     */
```

Comment:
Suggest either removing this or re-formatting this (so not to use doxygen `@retval` tags).

We don't want to have doxygent docs for `private` methods. It is fine to not document simple `private` methods.

Reply:
Okay. I have removed these comments.